internal/templates/containercapabilities/template.go

```diff
@@ -0,0 +1,218 @@
+) *diagnostic.Diagnostic {
+	if forbidAll {
+		// User has forbidden all capabilities
+		for _, scCap := range scCaps.Add {
+			var excluded bool
+			for _, exceptionCapMatcher := range exceptionCapMatchers {
+				if exceptionCapMatcher(string(scCap)) {
+					// Forgive this capability
+					excluded = true
+					break
+				}
+			}
+			if !excluded {
+				return &diagnostic.Diagnostic{
+					Message: fmt.Sprintf(
+						"container %q has ADD capability: %q, but no capabilities should be added at all and"+
+							" this capabilty is not included in the exceptions list",
+						containerName,
+						scCap),
+				}
+			}
+		}
+		// No violations
+		return nil
+	}
+
+	// Any capability from scCaps should not match with any from paramCaps
+	for _, paramCapMatcher := range paramCapMatchers {
+		for _, scCap := range scCaps.Add {
+			// User can specify to add "all" under containers as well.
+			if paramCapMatcher(string(scCap)) || literalReservedCapabilitiesAllMatcher(string(scCap)) {
+				// A capability from ADD list matched with a cap from forbidden capabilities list.
+				return &diagnostic.Diagnostic{
+					Message: fmt.Sprintf("container %q has ADD capability: %q, which matched with "+
+						"the forbidden capability for containers",
+						containerName,
+						scCap),
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func checkForbidAll(paramCaps []string) bool {
+	for _, paramCap := range paramCaps {
+		if literalReservedCapabilitiesAllMatcher(paramCap) {
+			return true
+		}
+	}
+	return false
+}
+
+func verifyExceptionsList(forbidAll bool, exceptions []string) error {
+	// Check if forbidAll is set
+	if !forbidAll && len(exceptions) != 0 {
+		return errors.Errorf("for verifying container capabilities, \"Exceptions\" list should only"+
+			" be filled when %q capabilities specified in the forbidden list", reservedCapabilitiesAll)
+	}
+	// Check no "all" in exceptions list
+	for _, cap := range exceptions {
+		if literalReservedCapabilitiesAllMatcher(cap) {
+			return errors.Errorf("capabilities exceptions list should not contain %q", reservedCapabilitiesAll)
+		}
+	}
+	return nil
+}
+
+func init() {
+	templates.Register(check.Template{
+		HumanName:   "Verify container capabilities",
+		Key:         "verify-container-capabilities",
+		Description: "Flag containers that do not match capabilities requirements",
+		SupportedObjectKinds: check.ObjectKindsDesc{
+			ObjectKinds: []string{objectkinds.DeploymentLike},
+		},
+		Parameters:             params.ParamDescs,
+		ParseAndValidateParams: params.ParseAndValidate,
+		Instantiate: params.WrapInstantiateFunc(func(p params.Params) (check.Func, error) {
+			var err error
+			literalReservedCapabilitiesAllMatcher, err = matcher.ForString(matchLiteralReservedCapabilitiesAll)
+			if err != nil {
+				return nil, err
+			}
+
+			// Check for "all" in case user does not want any capability in containers.
+			forbidAll := checkForbidAll(p.ForbiddenCapabilities)
```

Comment:
Should we also enforce that if `all` is in the list, then the list has no other elements?